### PR TITLE
mini cart component 

### PR DESCRIPTION
**Pull request by jatinkatyal13**

### CODE REVIEW

**Comment by abhishek97** [comment]
This pull request **introduces 1 alert** when merging 42672f2d15ada139a20a1570277faed6f9ed75c8 into 9acd3a8f88afc897752b16567b71490cb4e1e2d7 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-9bf5678fa6ee29ac29b211e0fcd2dc304a32048f)

**new alerts:**

* 1 for Useless assignment to local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by abhishek97** [comment]
This pull request **introduces 1 alert** when merging a37b08359607a16f212e15ef37c86ea91897870b into 9acd3a8f88afc897752b16567b71490cb4e1e2d7 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-1c56d1131873736df612e7e003bc2093b5c4cf39)

**new alerts:**

* 1 for Useless assignment to local variable

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: app/pods/components/cart-dialog/component.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  dukaanUrl = env.dukaanUrl
+
+  addCartTask = task(function *(){
+    yield this.get('api').request('/runs/clear_cart');
+    const runId = this.get('run.id');
+    yield this.get('api').request(`/runs/${runId}/buy`);
+    window.location.href = this.get('dukaanUrl')
+  })
+
+  didReceiveAttrs() {
```
Comment (abhishek97) [inline app/pods/components/cart-dialog/component.js]: This is useless

File: app/pods/components/cart-dialog/component.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    this.set('run', this.get('run'))
+    this.set('dukaanCart', this.get('dukaanCart'))
+  }
+
+  @action
+  onClose() {
```
Comment (abhishek97) [inline app/pods/components/cart-dialog/component.js]: Might as well just directly bind it in template. No need to define a wrapper action here.
Check this twiddle here: https://ember-twiddle.com/f82322578e9beedcb148986938b7ab78?openFiles=controllers.application.js%2Ctemplates.components.modal-comp.hbs

File: app/styles/app.scss [hunk 3]
```diff
@@ -52,4 +52,13 @@ body.centered-modal-showing {
 .modal-container {
   max-height: 80vh;
   overflow: auto;
-}
\ No newline at end of file
+}
+
+.ember-modal-overlay {
```
Comment (abhishek97) [inline app/styles/app.scss]: Don't modify ember-modal-overlay classes directly, Instead use your own custom classes.
See `custom styles` sections here: http://yapplabs.github.io/ember-modal-dialog/

Comment (jatinkatyal13) [reply app/styles/app.scss]: Actually, this what @vdvibhu20 asked me to do! But I'll update!

**Comment by abhishek97** [comment]
- Clicking Buy on empty cart should also redirect to dukaan.
- Adding the same product again should not show popup

**Comment by abhishek97** [comment]
@jatinkatyal13 Back merge with master and make sure build works

**Comment by abhishek97** [comment]
This pull request **introduces 2 alerts** when merging 66ef32d73da425f7feabbdcb6e322df888a11f75 into d6192607a8a32671eea5f958611f7ef1a99d5311 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-bfe249aa72fa13b3f0e5ca6629f05ff7a0e62f78)

**new alerts:**

* 2 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by abhishek97** [comment]
This pull request **introduces 1 alert** when merging bfc25e92465c2e50d31401feb030d90421c8b6b2 into 433a85a6be47b59d73b3ac183a335eff09d798b7 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-6002d392a622cc092bfb50599e583a7ddb89007b)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by abhishek97** [comment]
This pull request **introduces 1 alert** when merging b7e55a99a8cce9136b9e4e745fd505326127b527 into df0d04f65d4cd8dfaaabe3a4e4927bbfe43e8783 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-ef868ee1eacd5bdeb2d4078341bbf7b31c01b361)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by abhishek97** [comment]
This pull request **introduces 1 alert** when merging 83c706a4ba5d064716a78f3333ccdcfa8ba45554 into a9d1b47479fa1d1cac6e9635f1141d4e8eaf32b3 - [view on LGTM.com](https://lgtm.com/projects/g/coding-blocks/codingblocks.online.projectx/rev/pr-0b60a0f4dcc7342333caa5172a9228cf1ce5fa21)

**new alerts:**

* 1 for Syntax error

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Pull request merged**